test/e2e/util.go

```diff
@@ -430,7 +471,7 @@ func parseNetexecResponse(response string) (string, error) {
 	}
 	if len(res.Errors) > 0 {
 		if strings.Contains(strings.ToLower(res.Errors[0]), "timeout") {
-			return "Timeout", nil
+			return "", NewTimeoutError("timed out during curl: %v", res.Errors)
```

Comment:
Maybe we can just return an error here using `fmt.Errorf` since we don't really care if connection failed due to timeout or anything else, all the tests will just retry? It will simplify thing a little

Reply:
I don't know. I looked at the different changes and the initial commit was intended to fail immediately on any issue, here:
https://github.com/ovn-org/ovn-kubernetes/commit/753b5e391bab3f52809cc60959e5432239c7243b

The timeout exception/retry was added later on, here:
https://github.com/ovn-org/ovn-kubernetes/commit/61ae6572b3c88e16f6320e6cef5fdfca93856295

So the tests would iterate until maxtries was read and would try to hit all endpoints. However, for failure scenaios, they started with:
* Fail on any issue, immediately
Then were changed to:
* Fail on any issue, immediately, unless it's a timeout; in that case, do not fail immediately but return string "Timeout" (which was broken in pokeEndpointHostname anyway because the "Timeout" string would be parsed as an endpoint name and in the end would be compared to the list of endpoints and lead to a fail anyway --> we might as well have failed fast here).

So the simplification that you are suggesting would be:
* Retry on any issue, until we hit the number of maxtries.

If we now change this to try again on any issue, wouldn't we change what we are testing? If simply retrying is ok, we can indeed simplify it. But that would relax the test conditions. We can also see if we can go back to failing on anything, even if there's a timeout; however I suppose that @astoycos had a reason for making that timeout exception there.

I could also just catch the "Timeout" string in the poke methods but I don't think that's the right way of implementing error handling.
I could also just return an error with `fmt.Errorf("Timeout: %s", msg)`  and then rewrite `FailUnlessTimeoutError` to do a `string.Contains(err.Error(), "Timeout")`. That would get rid of the TimeoutError struct (https://github.com/ovn-org/ovn-kubernetes/pull/2702/commits/1c0fcdea39d756f7a16f85e04888b8a087d55d91#diff-c3b02e1c6c92fc59f6b49a20a38dbf177d8bf4189579639ab6e08a329b85cf59R80) but I don't think that this adds a lot of complexity and is cleaner than checking for a string.

I'll follow your suggestion, so just let me know what I should try and I'll go ahead with that.

Reply:
Done.